modules/platforms/dotnet/Apache.Ignite.Core/Impl/Cache/CacheImpl.cs

```diff
@@ -981,23 +992,39 @@ private Task<long> SizeAsync0(int? part, params CachePeekMode[] modes)
             var holder = new CacheEntryProcessorHolder(processor, arg,
                 (e, a) => processor.Process((IMutableCacheEntry<TK, TV>)e, (TArg)a), typeof(TK), typeof(TV));
 
-            return DoOutOpAsync(CacheOp.InvokeAsync, writer =>
-                {
-                    writer.WriteObjectDetached(key);
-                    writer.WriteObjectDetached(holder);
-                },
-                r =>
-                {
-                    if (r == null)
-                        return default(TRes);
+            var ptr = AllocateNoTx(holder);
+
+            try
+            {
+                return DoOutOpAsync(CacheOp.InvokeAsync, writer =>
+                    {
+                        writer.WriteObjectDetached(key);
+                        writer.WriteLong(ptr);
+                        writer.WriteObjectDetached(holder);
+                    },
+                    r =>
```

Comment:
Lets be consistent with the first arg, rename to "reader" (or "writer" to "w")
Here and below.

Reply:
Agree, fixed.